rust/src/lib/ifaces/inter_ifaces.rs

```diff
@@ -1017,6 +1019,117 @@ impl MergedInterfaces {
         Ok(())
     }
 
+    fn get_hsr_mac(&self, iface: &HsrInterface) -> Option<String> {
+        if iface.base.mac_address.is_some() {
+            return iface.base.mac_address.clone();
+        }
+
+        let hsr_conf = iface.hsr.as_ref()?;
+
+        for use_permanent in [false, true] {
+            for port in [&hsr_conf.port1, &hsr_conf.port2] {
+                let mac = self.kernel_ifaces.get(port).and_then(|iface| {
+                    let base = iface.merged.base_iface();
+                    if use_permanent {
+                        base.permanent_mac_address.clone()
+                    } else {
+                        base.mac_address.clone()
+                    }
+                });
+                if mac.is_some() {
+                    return mac;
+                }
+            }
+        }
+
+        None
+    }
+
+    fn get_desired_mac(&self, name: &str) -> Option<String> {
```

Comment:
You only searched `kernel_ifaces`, but function name does not reflect this, future developer might use this function in a wrong way.

Personally I don't think we should wrap this simple stuff into function.